cmd/gardener-metrics-exporter/app.go

```diff
@@ -122,30 +125,37 @@ func run(o *options, closeCh chan os.Signal) error {
 	return nil
 }
 
-func setupInformerFactories(kubeconfigPath string, stopCh <-chan struct{}) (gardeninformers.SharedInformerFactory, kubeinformers.SharedInformerFactory, error) {
-	k8sGardenClient, err := kubernetes.NewClientFromFile(kubeconfigPath)
+func newConfigFromBytes(kubeconfig string) *restclient.Config {
+	kubecf, err := ioutil.ReadFile(kubeconfig)
 	if err != nil {
-		return nil, nil, err
+		fmt.Println("Could not read config")
+		os.Exit(1)
 	}
+	configObj, err := clientcmd.Load(kubecf)
+	if err != nil {
+		fmt.Println("Could not load config")
+		os.Exit(1)
+	}
+	clientConfig := clientcmd.NewDefaultClientConfig(*configObj, &clientcmd.ConfigOverrides{})
+	client, err := clientConfig.ClientConfig()
```

Comment:
`client` can also be potentially nil. Either we catch this here and return an error or we have to check for nil object after calling `newConfigFromBytes()`.